geopandas/tests/test_geoseries.py

```diff
@@ -180,5 +182,17 @@ def test_proj4strings(self):
         reprojected_dict = self.g3.to_crs({'proj': 'utm', 'zone': '30N'})
         self.assertTrue(np.alltrue(reprojected_string.geom_almost_equals(reprojected_dict)))
 
+
+def test_construct_from_series():
+    shapes = [Polygon([(random.random(), random.random()) for _ in range(3)])
+              for _ in range(10)]
+    s = pd.Series(shapes, index=list('abcdefghij'), name='foo')
+    g = GeoSeries(s)
+
+    assert [a.equals(b) for a, b in zip(s, g)]
```

Comment:
Isn't this always True if the list is non-empty, even if the elements are False?